eul/Schema/EulComponent.swift

```diff
@@ -38,6 +38,14 @@ enum EulComponent: String, CaseIterable, Identifiable {
     case Memory
     case Battery
     case Network
+
+    static var allCases: [EulComponent] {
+        if BatteryStore().valid {
```

Comment:
1. no need for constructing a new instance here? just `BatteryStore.shared`
2. will it better to append battery component directly if needed, e.g.:

```swift
let components: [EulComponent] = [.CPU, .Fan, .Memory, .Network]
return BatteryStore.shared.isValid ? components + [.Battery] : components
```

Reply:
Great advice!